### ISSUE TITLE
automatic refresh not working with beta.11 ??

### ISSUE THREAD

**Issue by rhday74** [webpack/webpack-dev-server#707]
I've got an Angular 2.2.3 project using Webpack beta.27 on Mac OS X Sierra, Node version 6.9.1

I didn't change anything else in my webpack configuration (besides the obvious stuff that was broken between beta.25 and beta.27) but for some reason automatic refresh just stopped working with webpack-dev-server beta.11.  If I roll back to beta.10 it works fine, every code change is picked up, browser refreshes, etc.

Is there something new I have to configure with beta.11??  Just curious if anybody else is seeing this behavior

**Comment by SpaceK33z** [comment]
Can you be a bit more specific? What is your config like? Do you see any errors? Is it in all browsers?

**Comment by rhday74** [comment]
Yes of course.  Seeing this behavior on Chrome.  I haven't tried any other browsers yet.

Below is the base configuration:

```
const TEMP_PATH = '.tmp',

    // webpack instance and helper funcs
    WEBPACK = require('webpack'),
    HELPERS = require('./helpers'),

    // webpack plugins
    COPY_WEBPACK_PLUGIN           = require('copy-webpack-plugin'),
    SPRITE_SMITH_PLUGIN           = require('webpack-spritesmith'),
    FORK_CHECKER_PLUGIN           = require('awesome-typescript-loader').ForkCheckerPlugin,
    ASSETS_PLUGIN                 = require('assets-webpack-plugin'),
    CONTEXT_REPLACEMENT_PLUGIN    = require('webpack/lib/ContextReplacementPlugin'),

    /*
     * Webpack Constants
     */
    METADATA = {
        title           : 'LogistiCAD TNG UI',
        baseUrl         : '/',
        isDevServer     : HELPERS.isWebpackDevServer(),
        stylesConfig    : {
            browsers : [
                'Android 2.3',
                'Android >= 4',
                'Chrome >= 20',
                'Firefox >= 24',
                'Explorer >= 8',
                'iOS >= 6',
                'Opera >= 12',
                'Safari >= 6'
            ],
            spritePadding   : 10,
            spriteAlgorithm : 'binary-tree'
        }
    };

/*
 * Webpack configuration
 *
 * See: http://webpack.github.io/docs/configuration.html#cli
 */
module.exports = {
    /*
     * The entry point for the bundle
     * Our Angular.js app
     *
     * See: http://webpack.github.io/docs/configuration.html#entry
     */
    entry : {
        polyfills : './src/polyfills.browser.ts',
        vendor    : './src/vendor.browser.ts',
        main      : './src/main.browser.ts'
    },

    /*
     * Options affecting the resolving of modules.
     *
     * See: http://webpack.github.io/docs/configuration.html#resolve
     */
    resolve : {
        /*
         * An array of extensions that should be used to resolve modules.
         *
         * See: http://webpack.github.io/docs/configuration.html#resolve-extensions
         */
        extensions : ['.ts', '.js', '.json', '.scss'],

        // An array of directory names to be resolved to the current directory
        modules : [
            HELPERS.root('src'),
            'node_modules',
            'content',
            TEMP_PATH,
        ]
    },

    /*
     * Separate resolve options for loaders
     *
     * See: http://webpack.github.io/docs/configuration.html#resolve
     */
    resolveLoader : {
        /*
         * An array of extensions that should be used to resolve modules.
         *
         * See: http://webpack.github.io/docs/configuration.html#resolve-extensions
         */
        extensions : ['.ts', '.js', '.json', '.scss'],

        // An array of directory names to be resolved to the current directory
        modules : [
            HELPERS.root('src'),
            'node_modules',
            'content',
            TEMP_PATH,
        ]
    },

    /*
     * Options affecting the normal modules.
     *
     * See: http://webpack.github.io/docs/configuration.html#module
     */
    module : {
        /**
         * rules for modules (configure loaders, parser options, etc.)
         *
         * See: http://webpack.js.org/configuration/module/#module-rules
         */
        rules : [
            /*
             * Tslint loader support for *.ts files
             *
             * See: https://github.com/wbuchwalter/tslint-loader
             */
            {
                test    : /\.ts$/,
                loader  : 'tslint-loader',
                enforce : 'pre',
            },

            /*
             * Source map loader support for *.js files
             * Extracts SourceMaps for source files that as added as sourceMappingURL comment.
             *
             * See: https://github.com/webpack/source-map-loader
             */
            {
                test    : /\.js$/,
                loader  : 'source-map-loader',
                exclude : [
                    // these packages have problems with their sourcemaps
                    HELPERS.root('node_modules/rxjs'),
                    HELPERS.root('node_modules/@ngrx/core'),
                    HELPERS.root('node_modules/@angular')
                ],
                enforce : 'pre'
            },

            /*
             * Typescript loader support for .ts files
             *
             * See: https://github.com/s-panferov/awesome-typescript-loader
             */
            {
                test    : /\.ts$/,
                loaders : [
                    'awesome-typescript-loader',
                    'angular2-template-loader'
                ],
                exclude : [/\.(spec|e2e)\.ts$/]
            },

            /*
             * Json loader support for *.json files.
             *
             * See: https://github.com/webpack/json-loader
             */
            {
                test    : /\.json$/,
                loader  : 'json-loader'
            },

            /**
             * support for .txt files
             */
            {
                test    : /\.txt/,
                loader  : 'raw-loader'
            },

            /**
             * all sass required in component files will be merged in js files
             */
            {
                test    : /\.component\.scss$/,
                loaders : [
                    'raw-loader',
                    'postcss-loader',
                    'sass-loader'
                ]
            },

            /**
             * handle individual svg files that are not fonts
             * these svg files will be optimized and inlined
             * as utf-8 encoded DataUrl strings in the HTML
             *
             * Reference: https://github.com/bhovhannes/svg-url-loader
             */
            {
                test    : /\.svg$/,
                loader  : 'svg-url-loader',
                exclude  : [
                    HELPERS.root('src/app/content/fonts'),
                    HELPERS.root('src/app/content/images/SvgIcons')
                ],
            },

            /* Raw loader support for *.html
             * Returns file content as string
             *
             * See: https://github.com/webpack/raw-loader
             */
            {
                test    : /\.html$/,
                loader  : 'raw-loader',
                exclude : [HELPERS.root('src/index.html')]
            }
        ]
    },

    /*
     * Add additional plugins to the compiler.
     *
     * Reference: http://webpack.github.io/docs/configuration.html#plugins
     */
    plugins : [
        /*
         * Plugin: AssetsPlugin
         * Description: When working with Webpack you might want to generate your bundles with a generated hash in them (for cache busting).
         *              This plug-in outputs a json file with the paths of the generated assets so you can find them from somewhere else.
         *              Emits a json file with assets paths.
         *
         * See: https://github.com/kossnocorp/assets-webpack-plugin
         */
        new ASSETS_PLUGIN({
            path        : HELPERS.root('.dist'),
            filename    : 'webpack-assets.json',
            prettyPrint : true
        }),

        /*
         * Plugin: ForkCheckerPLugin
         * Description: Do type checking in a separate process, so webpack don't need to wait.
         *
         * See: https://github.com/s-panferov/awesome-typescript-loader#forkchecker-boolean-defaultfalse
         */
        new FORK_CHECKER_PLUGIN(),

        /*
         * Plugin: CommonsChunkPlugin
         * Description: Shares common code between the pages.
         * It identifies common modules and put them into a commons chunk.
         *
         * See: https://webpack.github.io/docs/list-of-plugins.html#commonschunkplugin
         * See: https://github.com/webpack/docs/wiki/optimization#multi-page-app
         */
        new WEBPACK.optimize.CommonsChunkPlugin({
            name : ['main', 'vendor', 'polyfills']
        }),

        /**
         * Plugin: ContextReplacementPlugin
         * Description: Provides context to Angular's use of System.import
         *
         * See: https://webpack.github.io/docs/list-of-plugins.html#contextreplacementplugin
         * See: https://github.com/angular/angular/issues/11580
         */
        new CONTEXT_REPLACEMENT_PLUGIN(
            // The (\\|\/) piece accounts for path separators in *nix and Windows
            /angular(\\|\/)core(\\|\/)(esm(\\|\/)src|src)(\\|\/)linker/,
            HELPERS.root('src') // location of your src
        ),

        /*
         * Plugin: CopyWebpackPlugin
         * Description: Copy files and directories in webpack.
         *
         * Copies project static assets.
         *
         * See: https://www.npmjs.com/package/copy-webpack-plugin
         */
        new COPY_WEBPACK_PLUGIN([{
            from    : 'src/app/content',
            to      : 'content'
        }], {
            ignore  : [
                // don't copy any files with .scss extension
                '*.scss',

                // ignore all svg icons
                'images/SvgIcons',
                {
                    glob    : 'styles/*.*',
                    dot     : true
                },
                {
                    glob    : 'images/SvgIcons/*.*',
                    dot     : true
                }
            ]
        }),

        /**
         * plugin that converts set of images into a spritesheet and
         * SASS/LESS/Stylus mixins, using spritesmith and spritesheet-templates
         *
         * Reference: https://github.com/mixtur/webpack-spritesmith
         */
        new SPRITE_SMITH_PLUGIN({
            src : {
                cwd : HELPERS.root('src/app/content/images/sprite'),
                glob    : '*.*'
            },
            target : {
                image   : './' + TEMP_PATH + '/content/sprite.png',
                css     : './' + TEMP_PATH + '/_sprite.scss'
            },
            apiOptions : {
                cssImageRef : '~content/sprite.png'
            },
            spritesmithOptions : {
                padding     : METADATA.stylesConfigspritePadding,
                algorithm   : METADATA.stylesConfigspriteAlgorithm
            }
        }),
    ],

    /*
     * Include polyfills or mocks for various node stuff
     * Description: Node configuration
     *
     * See: https://webpack.github.io/docs/configuration.html#node
     */
    node : {
        global          : true,
        crypto          : 'empty',
        process         : true,
        module          : false,
        clearImmediate  : false,
        setImmediate    : false
    }
};
```

And this is the development / local debugging config that piggyback's off the above common config:

```
const DIST_PATH   = __dirname + '/.dist',

    HELPERS = require('./helpers'),

    WEBPACK = require('webpack'),

    // used to merge webpack configs
    WEBPACK_MERGE = require('webpack-merge'),

    // the settings that are common to all webpack builds
    COMMON_CONFIG = require('./webpack.common.js'),

    /**
     * Webpack Plugins
     */
    DEFINE_PLUGIN        = require('webpack/lib/DefinePlugin'),
    NAMED_MODULES_PLUGIN  = require('webpack/lib/NamedModulesPlugin'),
    HTML_WEBPACK_PLUGIN   = require('html-webpack-plugin'),
    EXTRACT_TEXT_PLUGIN   = require('extract-text-webpack-plugin'),

    /**
     *  PostCSS plugins
     * @type {any}
     */
    AUTO_PREFIXER    = require('autoprefixer'),
    POST_CSS_ASSETS   = require('postcss-assets'),
    POST_CSS_MQ_PACKER = require('css-mqpacker'),

    /**
     * Environment Targeting
     * Get npm lifecycle event to identify the environment
     */
    ENV = process.env.npm_lifecycle_event,

    /*
     * Webpack Constants
     */
    METADATA = WEBPACK_MERGE(COMMON_CONFIG.metadata, {
        title           : 'LogistiCAD TNG UI',
        baseUrl         : '/',
        isDevServer     : HELPERS.isWebpackDevServer(),
        stylesConfig    : {
            browsers : [
                'Android 2.3',
                'Android >= 4',
                'Chrome >= 20',
                'Firefox >= 24',
                'Explorer >= 8',
                'iOS >= 6',
                'Opera >= 12',
                'Safari >= 6'
            ],
            spritePadding   : 10,
            spriteAlgorithm : 'binary-tree'
        },
        API_ROOT    : 'https://gw.devcloud.logisticad.com:8443/',
        API_CONFIG  : {
            /* dropdown types */
            getMetaDataTypes    : 'lcad/v1/person/metadataTypes',

            /* beneficiary */
            getBeneficiary   : 'getBeneficiary/',

            /* eligibility */
            getEligibility   : 'eligibility/',

            /* reservation */
            getReservation   : 'reservation/',

            /* dashboard */
            getDashBoard     : 'dashboard/',

            /* search types api */
            getSearchTypes : 'searchTypes',

            /* new beneficiary profile */
            createBeneficiaryProfile   : 'createBeneficiaryProfile/'
        }
    });

/**
 * Webpack configuration
 *
 * See: http://webpack.github.io/docs/configuration.html#cli
 */
module.exports = WEBPACK_MERGE(COMMON_CONFIG, {
    /**
     * Developer tool to enhance debugging
     *
     * See: http://webpack.github.io/docs/configuration.html#devtool
     * See: https://github.com/webpack/docs/wiki/build-performance#sourcemaps
     */
    devtool : 'cheap-module-eval-source-map',  // set to 'source-map' or 'cheap-module-source-map' to view root SASS sourcemaps

    /*
     * Output
     * Reference: http://webpack.github.io/docs/configuration.html#output
     * Should be an empty object if it's generating a test build
     * Karma will handle setting it up for you when it's a test build
     */
    output : {
        /**
         * The output directory as absolute path (required).
         *
         * Reference: http://webpack.github.io/docs/configuration.html#output-path
         */
        path : HELPERS.root('.dist'),

        /**
         * Specifies the name of each output file on disk.
         * IMPORTANT: You must not specify an absolute path here!
         *
         * Reference: http://webpack.github.io/docs/configuration.html#output-filename
         */
        filename : '[name].bundle.js',

        /**
         * The filename of the SourceMaps for the JavaScript files.
         * They are inside the output.path directory.
         *
         * Reference: http://webpack.github.io/docs/configuration.html#output-sourcemapfilename
         */
        sourceMapFilename : '[name].map',

        /**
         * The filename of non-entry chunks as relative path
         * inside the output.path directory.
         *
         * Reference: http://webpack.github.io/docs/configuration.html#output-chunkfilename
         */
        chunkFilename : '[id].chunk.js'
    },

    /*
     * Options affecting the normal modules.
     *
     * See: http://webpack.github.io/docs/configuration.html#module
     */
    module : {
        /*
         * An array of automatically applied loaders.
         *
         * IMPORTANT: The loaders here are resolved relative to the resource which they are applied to.
         * This means they are not resolved relative to the configuration file.
         *
         * See: http://webpack.github.io/docs/configuration.html#module-loaders
         */
        rules : [
            /**
             * all css in sass-config.scss will be bundled in an external css file
             */
            {
                test    : /\.scss$/,
                include : [
                    HELPERS.root('src/app/sass-config.scss'),
                    HELPERS.root('src/app/styles')
                ],
                loader  : EXTRACT_TEXT_PLUGIN.extract({
                    loader         : [
                        {
                            loader  : 'css-loader',
                            query   : {
                                sourceMap : true
                            }
                        },
                        'postcss-loader',
                        {
                            loader  : 'sass-loader',
                            query   : {
                                sourceMap : true
                            }
                        }
                    ],
                    fallbackLoader : 'style-loader'
                })
            },

            /**
             * copy fonts to output
             * excluding images directory since svg can be images or fonts
             */
            {
                test    : /\.(eot|woff|woff2|ttf|svg)(\?\S*)?$/,
                loader  : 'file-loader',
                query   : {
                    name : 'content/fonts/[name].[ext]'
                },
                exclude : [HELPERS.root('src/app/content/images')],
            },

            /**
             * minify and copy images to output
             */
            {
                test    : /\.(png|jpe?g|gif)(\?\S*)?$/,
                loaders : [
                    {
                        loader  : 'file-loader',
                        query   : {
                            name : 'content/images/[name].[ext]'
                        }
                    }
                ],
                exclude : [
                    /node_modules/,
                    HELPERS.root('src/app/content/fonts')
                ]
            },

            /**
             * Any svg files required from the `content/images/SvgIcons` directory
             * will be combined into an svg store that gets added to the DOM
             * and referenced in a `<use>` tag in the HTML
             *
             * Reference: https://github.com/kisenka/svg-sprite-loader
             */
            {
                test    : /\.svg$/,
                include : [HELPERS.root('src/app/content/images/SvgIcons')],
                loader  : 'svg-sprite-loader?' + JSON.stringify({
                    name        : '[name]',
                    prefixize   : true
                })
            }
        ]
    },

    /*
     * Add additional plugins to the compiler.
     *
     * Reference: http://webpack.github.io/docs/configuration.html#plugins
     */
    plugins : [
        /**
         * Plugin: DefinePlugin
         * Description: Define free variables.
         * Useful for having development builds with debug logging or adding global constants.
         *
         * See: https://webpack.github.io/docs/list-of-plugins.html#defineplugin
         *
         * NOTE: when adding more properties make sure you include them in custom-typings.d.ts
         */
        new DEFINE_PLUGIN({
            title       : JSON.stringify(METADATA.title),
            baseUrl     : JSON.stringify(METADATA.baseUrl),
            isDevServer : JSON.stringify(METADATA.isDevServer),
            ENV         : JSON.stringify(ENV),
            API_ROOT    : JSON.stringify(METADATA.API_ROOT),
            API_CONFIG  : JSON.stringify(METADATA.API_CONFIG)
        }),

        /**
         * Plugin: LoaderOptionsPlugin
         * Description : The UglifyJsPlugin no longer puts loaders into minimize mode. The debug option has been removed.
         * Loaders should no longer read their options from the webpack configuration.
         * Instead you need to provide these options with the LoaderOptionsPlugin.
         */
        new WEBPACK.LoaderOptionsPlugin({
            minimize    : false,
            debug       : true,
            options     : {
                context : __dirname,
                tslint : {
                    /**
                     * tslint errors are displayed by default as warnings
                     * set emitErrors to true to display them as errors
                     */
                    emitErrors : false,

                    /**
                     * tslint does not interrupt the compilation by default
                     * if you want any file with tslint errors to fail
                     * set failOnHint to true
                     */
                    failOnHint : false,

                    /**
                     * files to target for linting live here
                     */
                    resourcePath : './src'
                },

                /**
                 * PostCSS Loader Configuration
                 * Reference: https://github.com/geut/postcss-copy
                 * Copy images referenced in component stylesheets
                 *
                 * Reference: https://github.com/TrySound/postcss-inline-svg
                 * Inline SVG and customize its styles
                 *
                 * Reference: https://github.com/assetsjs/postcss-assets
                 * Asset management for component CSS
                 *
                 * Reference: https://github.com/postcss/autoprefixer-core
                 * Add vendor prefixes to your css
                 */
                postcss : function () {
                    return [
                        POST_CSS_ASSETS({
                            basePath    : './src/app/content',
                            baseUrl     : 'content',
                            loadPaths   : [HELPERS.root(DIST_PATH, 'content')]
                        }),
                        POST_CSS_MQ_PACKER(),
                        AUTO_PREFIXER({
                            browsers    : METADATA.stylesConfig.browsers
                        })
                    ];
                },

                /*
                 * Sass-Loader
                 * Description: pass options to node-sass with sassLoader property
                 * See: https://github.com/jtangelder/sass-loader
                 */
                sassLoader : {
                    includePaths : [
                        HELPERS.root('src/app/styles'),
                        HELPERS.root('src/app/shared/components'),
                        HELPERS.root('src/app/areas')
                    ]
                }
            }
        }),

        /**
         * Plugin: NamedModulePlugin (experimental)
         * Description: Uses file names as module name.
         *
         * See: https://github.com/webpack/webpack/commit/a04ffb928365b19feb75087c63f13cadfc08e1eb
         */
        new NAMED_MODULES_PLUGIN(),

        /*
         * Plugin: ExtractTextPlugin
         * Description: Extract css from inline styles in the JS bundles and create separate CSS bundles
         *
         * See: https://github.com/webpack/extract-text-webpack-plugin
         */
        new EXTRACT_TEXT_PLUGIN({
            filename    : 'style.css',
            allChunks   : true
        }),

        /*
         * Plugin: HTMLWebpackPlugin
         * Description: Simplifies creation of HTML files to serve your webpack bundles.
         * This is especially useful for webpack bundles that include a hash in the filename
         * which changes every compilation.
         *
         * See: https://github.com/ampedandwired/html-webpack-plugin
         */
        new HTML_WEBPACK_PLUGIN({
            template        : 'src/index.html',
            chunksSortMode  : 'dependency'
        })
    ],

    /**
     * webpack-dev-server configuration
     * Reference: http://webpack.github.io/docs/configuration.html#devserver
     * Reference: http://webpack.github.io/docs/webpack-dev-server.html
     */
    devServer : {
        contentBase         : HELPERS.root(DIST_PATH),
        quiet               : true,
        port                : 3000,
        host                : 'localhost',
        historyApiFallback  : true,
        watchOptions        : {
            aggregateTimeout    : 300,
            poll                : 1000
        }
    }
});
```

**Comment by rhday74** [comment]
I just tried it in Firefox and Safari and saw the same behavior.  I think i see what it is though.  I've got linting warnings apparently on recompile and that's preventing the reload.  Is there a setting that would let the reload happen inspite of those warnings??

**Comment by SpaceK33z** [comment]
Oh, reloading on warning was fixed again but hasn't been released yet. I'll do a release soon.

**Comment by SpaceK33z** [comment]
Fix released in v2.1.0-beta.12.
